localstack-core/localstack/services/lambda_/event_source_mapping/pollers/stream_poller.py

```diff
@@ -182,9 +194,12 @@ def poll_events_from_shard(self, shard_id: str, shard_iterator: str):
 
         # Send failed events to potential DLQ
         abort_condition = abort_condition or "RetryAttemptsExhausted"
+        # TODO: fix format for ESM using `requestContext` and `responseContext` (which requires info from the Lambda sender!) instead
         context = {
-            "condition": abort_condition,
-            "partnerResourceArn": self.partner_resource_arn,
+            "context": {
+                "condition": abort_condition,
+                "partnerResourceArn": self.partner_resource_arn,
+            }
```

Comment:
So this will soon differ depending on ESM/pipes, right?

Reply:
Correct, that's currently a parity issue in ESM v2. Difference:

```py
# ESM has this extra
"requestContext": {
  "requestId": "<uuid:2>",
  "functionArn": "arn:<partition>:lambda:<region>:111111111111:function:<resource:3>",
  "condition": "RetryAttemptsExhausted",
  "approximateInvokeCount": 2
},
"responseContext": {
  "statusCode": 200,
  "executedVersion": "$LATEST",
  "functionError": "Unhandled"
},

# Pipes has this extra
"context": {
    "condition": "RetryAttemptsExhausted",
    "partnerResourceArn": "arn:<partition>:pipes:<region>:111111111111:pipe/<pipe-name>"
  },
  ```